datastore/core/binarystore.py

```diff
@@ -256,6 +256,33 @@ class Datastore(trio.abc.AsyncResource):
 		return await (await self.get(key)).collect()
 	
 	
+	async def rename(self, key1: key_.Key, key2: key_.Key, *, replace: bool = True) -> None:
+		"""Moves the data at name *key1* to *key2*
+		
+		Arguments
+		---------
+		key1
+			The key to rename, must exist
+		key2
+			The new name of the key, if *replace* is ``False`` this must not exist
```

Comment:
Actually it's supposed to mean that the target key may not already exist if the *replace* parameter is False (cause if it did exist it would have to be replaced). As a matter of fact, this parameter *may* not be `None`, that wouldn't make any sense for a rename :upside_down_face:.

Do have any idea for a concise description for capturing the above meaning?

Reply:
Oh. Right.  
Can we phrase it as: "The new name of the key. If *replace* is ``False``, a key of the same name should not already exist"?

Reply:
Thank you! Applied wording similar to yours.